packages/core/src/database/migration.ts

```diff
@@ -0,0 +1,79 @@
+import { readdir, readFile } from 'fs/promises';
+import path from 'path';
+
+import chalk from 'chalk';
+import { DatabasePool, sql } from 'slonik';
+import { raw } from 'slonik-sql-tag-raw';
+
+const tablePath = 'node_modules/@logto/schemas/tables/migrations.sql';
+const migrationFilesDirectory = 'node_modules/@logto/schemas/migrations';
+
+const isMigrationTableExists = async (pool: DatabasePool) => {
+  const [result] = await pool.anyFirst<[boolean]>(sql`
+    select exists (
+      select from 
+        pg_tables
+      where 
+        tablename = '_migrations'
+      );
+  `);
+
+  return result;
+};
+
+const createMigrationTableIfNeeded = async (pool: DatabasePool): Promise<boolean> => {
+  if (await isMigrationTableExists(pool)) {
+    return false;
+  }
+
+  const query = await readFile(tablePath, 'utf-8');
+  await pool.query(sql`${raw(query)}`);
+
+  return true;
+};
+
+type MigrationScript = { up: (pool: DatabasePool) => Promise<void> };
+
+const getMigrationFiles = async () => {
+  const directory = await readdir(migrationFilesDirectory);
+  const files = directory.filter((file) => file.endsWith('.js'));
```

Comment:
will we keep other files under this migration path?

Reply:
No, only migration files.

Reply:
Though, we can add some file name check.